src/autolabel/transforms/ocr.py

```diff
@@ -1,110 +1,199 @@
-from typing import Dict, Any, List
+"""OCR transform for extracting text from documents."""
 
-from autolabel.transforms.schema import TransformType
-from autolabel.transforms import BaseTransform
-from autolabel.cache import BaseCache
+from __future__ import annotations
 
-from autolabel.transforms.schema import TransformError, TransformErrorType
+import io
+from pathlib import Path
+from typing import Any, ClassVar
+
+import boto3
+from PIL import Image
+
+from autolabel.cache import BaseCache
+from autolabel.transforms import BaseTransform
+from autolabel.transforms.schema import (
+    TransformError,
+    TransformErrorType,
+    TransformType,
+)
 
 
 class OCRTransform(BaseTransform):
-    """This class is used to extract text from any document using OCR. The output columns dictionary for this class should include the keys 'content_column' and 'metadata_column'
+    """Extract text from documents using OCR.
+
+    The output columns dictionary for this class should include the keys 'content_column'
+    and 'metadata_column'.
 
-    This transform supports the following image formats: PDF, PNG, JPEG, TIFF, JPEG 2000, GIF, WebP, BMP, and PNM
+    This transform supports the following image formats: PDF, PNG, JPEG, TIFF, JPEG 2000,
+    GIF, WebP, BMP, and PNM.
     """
 
-    COLUMN_NAMES = ["content_column"]
+    COLUMN_NAMES: ClassVar[list[str]] = ["content_column"]
 
     def __init__(
         self,
         cache: BaseCache,
-        output_columns: Dict[str, Any],
+        output_columns: dict[str, Any],
         url_column: str,
-        lang: str = None,
+        lang: str | None = None,
     ) -> None:
+        """Initialize OCRTransform.
+
+        Args:
+            cache: Cache instance to use
+            output_columns: Dictionary mapping output column names
+            url_column: Column containing document URLs/paths
+            lang: Optional language for OCR
+            custom_image_ocr: Optional custom OCR function
+
+        """
         super().__init__(cache, output_columns)
         self.url_column = url_column
         self.lang = lang
-
         try:
-            from PIL import Image
             import pytesseract
             from pdf2image import convert_from_path
 
-            self.Image = Image
             self.pytesseract = pytesseract
             self.convert_from_path = convert_from_path
             self.pytesseract.get_tesseract_version()
+
```

Comment:
probably dont need pytesseract for ocr now?

Reply:
Removed